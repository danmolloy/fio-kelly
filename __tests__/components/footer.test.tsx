import { render, screen, fireEvent, act } from "@testing-library/react";
import "@testing-library/jest-dom"
import Footer from "../../components/footer";
import "react-scroll"
import { animateScroll } from "react-scroll";

jest.mock("react-scroll", () => ({
  animateScroll: {
    scrollToTop: jest.fn()
  }
}))

describe("<Footer />", () => {
  beforeEach(() => {
    render(<Footer />)
  })
  it("footer-section is in the document", () => {
    const footer = screen.getByTestId("footer-section")
    expect(footer).toBeInTheDocument()
  })
  it("twitter link is in the document with expected href", () => {
    const twitterLink = screen.getByTitle("Link to Fiona's Twitter page")
    expect(twitterLink).toBeInTheDocument()
    expect(twitterLink).toHaveAttribute("href", "https://twitter.com/fizzlekelly")
  })
  it("'Back to top' btn is in the document and calls scrollToTop on click", async () => {
    const backToTop = screen.getByText("Back to top")
    expect(backToTop).toBeInTheDocument()
    await act(async() => {
      await fireEvent.click(backToTop)
    })
    expect(animateScroll.scrollToTop).toHaveBeenCalledWith({
      smooth: true, 
      duration: 750
    })
  })
  it("'Purple Rasberry Photography' link is in the document with expected href", () => {
    const photographerLink = screen.getByText("Purple Rasberry Photography")
    expect(photographerLink).toBeInTheDocument()
    expect(photographerLink).toHaveAttribute("href", "https://www.purpleraspberry.uk/")
    expect(photographerLink).toHaveAttribute("target", "_blank")
  })
  it("matches snapshot", () => {
    const footer = screen.getByTestId("footer-section")
    expect(footer).toMatchSnapshot()
  })
})