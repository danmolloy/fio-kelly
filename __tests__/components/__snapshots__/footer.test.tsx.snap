// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<section
  class="z-10 h-24 bottom-0 flex flex-row items-center justify-center border-t border-t-zinc-300 bg-zinc-100"
  data-testid="footer-section"
>
  <div
    class="flex flex-col items-center justify-center"
  >
    <button
      class="hover:underline font-display text-sm font-light hover:cursor-pointer active:text-zinc-400"
    >
      Back to top
    </button>
    <a
      class="text-sm font-light font-display self-end mt-3 text-gray-600 underline hover:text-blue-500"
      href="https://www.purpleraspberry.uk/"
      target="_blank"
    >
      Purple Rasberry Photography
    </a>
  </div>
</section>
`;
